docs/V2/TokenizationAndCompilation.md

```diff
@@ -0,0 +1,253 @@
+Old Replacement: `use **NAMESPACE_PREFIX_TOKEN****NAMESPACE_RELATIVE_TOKEN**\**PRIMARY_ACTOR_SHORT_PASCAL_CASE_NAME_TOKEN**`
+New Template: `use Neighborhoods\BuphaloTemplateTree\RelativeNamespace\PrimaryActorName`
+New Replacement: `use **TOKEN_NAMESPACE_PREFIX****TOKEN_NAMESPACE_RELATIVE\**TOKEN_PRIMARY_ACTOR_NAME_SHORT_PASCAL**`
+
+Old Template: `use Neighborhoods\BuphaloTemplateTree`
+Old Replacement: `use **NAMESPACE_PREFIX_TOKEN**`
+New Template: `use Neighborhoods\BuphaloTemplateTree`
+New Replacement: `use **TOKEN_NAMESPACE_PREFIX**`
+
+Old Template: `services:\n  Neighborhoods\BuphaloTemplateTree\PrimaryActorName`
+Old Replacement: `services:\n  **NAMESPACE_PREFIX_TOKEN****NAMESPACE_RELATIVE_TOKEN**\**PRIMARY_ACTOR_SHORT_PASCAL_CASE_NAME_TOKEN**`
+New Template: `services:\n  Neighborhoods\BuphaloTemplateTree\RelativeNamespace\PrimaryActorName`
+New Replacement: `services: **TOKEN_NAMESPACE_PREFIX****TOKEN_NAMESPACE_RELATIVE**\**TOKEN_PRIMARY_ACTOR_NAME_SHORT_PASCAL**`
+
+Old Template: `class: Neighborhoods\BuphaloTemplateTree\PrimaryActorName`
+Old Replacement: `class: **NAMESPACE_PREFIX_TOKEN****NAMESPACE_RELATIVE_TOKEN**\**PRIMARY_ACTOR_SHORT_PASCAL_CASE_NAME_TOKEN**`
+New Template: `class: Neighborhoods\BuphaloTemplateTree\RelativeNamespace\PrimaryActorName`
+New Replacement: `class: **TOKEN_NAMESPACE_PREFIX****TOKEN_NAMESPACE_RELATIVE**\**TOKEN_PRIMARY_ACTOR_NAME_SHORT_PASCAL**`
+
+Old Template: `['@Neighborhoods\BuphaloTemplateTree\PrimaryActorName`
+Old Replacement: `['@**NAMESPACE_PREFIX_TOKEN****NAMESPACE_RELATIVE_TOKEN**\**PRIMARY_ACTOR_SHORT_PASCAL_CASE_NAME_TOKEN**`
+New Template: `['@Neighborhoods\BuphaloTemplateTree\RelativeNamespace\PrimaryActorName`
+New Replacement: `['@**TOKEN_NAMESPACE_PREFIX****TOKEN_NAMESPACE_RELATIVE**\**TOKEN_PRIMARY_ACTOR_NAME_SHORT_PASCAL**`
+
+Old Template: `\PrimaryActorName;`
+Old Replacement: `**EMPTY_TOKEN**;`
+New Template: Not Applicable
+New Replacement: Not Applicable
+
+Old Template: `\{Template File Name};`
+Old Replacement: `**EMPTY_TOKEN**;`
+New Template: Not Applicable
+New Replacement: Not Applicable
+
+Old Template: `\PrimaryActorNameInterface`
+Old Replacement: `\**PRIMARY_ACTOR_SHORT_PASCAL_CASE_NAME_TOKEN**Interface`
+New Template: `\PrimaryActorNameInterface`
+New Replacement: `\**TOKEN_PRIMARY_ACTOR_NAME_SHORT_PASCAL**Interface`
+
+Old Template: `\{Template File Name}Interface`
+Old Replacement: `\**SHORT_PASCAL_CASE_NAME_TOKEN**Interface`
+New Template: Not Applicable
+New Replacement: Not Applicable
+
+Old Template: `\PrimaryActorName\`
+Old Replacement: `**EMPTY_TOKEN**\`
+New Template: Not Applicable
+New Replacement: Not Applicable
+
+Old Template: `\PrimaryActorName`
+Old Replacement: `\**RELATIVE_CLASS_PATH_TOKEN**`
+New Template: TODO
+New Replacement:TODO
+
+Old Template: `\{Template File Name}`
+Old Replacement: `\**SHORT_PASCAL_CASE_NAME_TOKEN**`
+New Template: Not Applicable
+New Replacement: Not Applicable
+
+Old Template: `protected $PrimaryActorName`
```

Comment:
is the `protected` part just an example? will this also work for `public` and `private`?

Reply:
Yeah, that's part of the old replacement search.